### DISCUSSION TITLE
[Idea] Add a redirection function 301 or 302 on Category page

### DISCUSSION THREAD

**Discussion by Patrickclicweb** [PrestaShop/PrestaShop#32269] [Ideas]
#### Is your feature request related to a problem?
It is a problem for SEO not to have this function add redirection function 301 or 302 on category page
I am always frustrated when I see that Prestashop has set up on the product pages the 301 or 302 redirects to another product or category and that this has not been done also for the category pages

#### Describe the solution you'd like
We could take a large part of the code of the product page to adapt it to the category page and to redirect 301 or 302 to another category page or to choose why not a CMS page in order to keep the SEO referencing of this category

#### Alternatives you've considered
For me I manage this in the .htaccess file, but it is not practical for the end customer

#### Additional context
Version 1,7.X,X

**Comment by florine2623** [comment]
Hello @Patrickclicweb ,

I edited your Feature request issue. Please edit it if needed or if information is missing.

Thank you for your suggestion. The Product Team will take it into consideration for future developments. 

Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know!

Thank you

**Comment by Patrickclicweb** [comment]
Hi,
the possibility of 301 or 302 redirection should be implemented in the .tpl of Category pages
as is now the case for the .tpl of Product pages

**Comment by jf-viguier** [comment]
Nice idea @Patrickclicweb.
Redirection could be a 310 too.
Can you explain the .tpl stuff please ?

**Comment by florine2623** [comment]
Hello all,

This has been added to the product page with this [issue](https://github.com/PrestaShop/PrestaShop/issues/23533) and [related PR](https://github.com/PrestaShop/PrestaShop/pull/29720) : 

https://user-images.githubusercontent.com/16019289/220396849-bb5fdd9a-a7f2-435d-8abc-97feb739a38a.mov

So I will close this issue.

Thanks!

**Comment by jf-viguier** [comment]
@florine2623 please reopen the issue, I"talking about the category page, the issue you mention is about product page.

**Comment by ghost** [comment]
Hello @florine2623 

Patrick was talking about categories and not products, this is a good idea for SEO.

I'm reopening the issue, don't hesitate to tell me if I didn't do well.

**Comment by Patrickclicweb** [comment]
Thank @okom3pom ! 
it is indeed from an SEO perspective

**Comment by MatShir** [comment]
Done by https://github.com/PrestaShop/PrestaShop/issues/35500
